### PR TITLE
Neurospit Tweak: Stamina Damage Nerfed, Weak Neurogas Spawn and Neurotoxin Reagent Transfer Added

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Neurospit reworked to output much lower Stamina damage, transfer neurotoxin chems to the target (subject to bio armor) and create weak, translucent neurotoxin gas.

Full neurotoxin gas (such as from the Defiler and Boiler) now extinguishes fire and burning mobs.

## Why It's Good For The Game

Eliminates the last remaining source of spike Stamina damage that can be exploited to easily secure captures, replacing it with more tactically and mechanically interesting area denial smoke and reagent transfer that imposes deferred/delayed Stamina damage, and acts as an ongoing debuff/pre-med counter. Also gas masks are now much more useful.

## Changelog
:cl:
tweak: Stamina damage dealt by neurospit reduced to 20 across the board. 
tweak: Neurospit now spawns a small area of weak neuro gas; strength of neurogas scales with the user's upgrade level. Praetorian neurospit gas area is larger.
tweak: Defiler and Boiler (full) neurogas extinguishes fire and burning mobs.
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
Most marine bio armour sits at 35%-50% , administering 3.5U neurotox or 4.5U neurotox per spit seems way too much and might make sentinels/spitter/praetorians too suppresive for anyone that might try to push,might be a good idea to tone it down to 1U per spit at weakest and 3U at strongest.

**Comment by Surrealaser** [comment]
> Most marine bio armour sits at 35%-50% , administering 3.5U neurotox or 4.5U neurotox per spit seems way too much and might make sentinels/spitter/praetorians too suppresive for anyone that might try to push,might be a good idea to tone it down to 1U per spit at weakest and 3U at strongest.
> Or you could make it more obvious that people are under the effect of neurotox and let them remove it by simple means like drinking water, making dylovene 10 times more effective against neurotox or making it metabolize away faster if marines are resting.

Given the hallucinogenic overlay, it's pretty obvious when you're affected by neurotox.

At 35% bio armor, only the Prae is going to come close to 5U+ neuro, but yeah, I'm open to tweaking the values downward. Keep in mind that neuro metas fast, 0.4 / sec, and its effects are very slow acting and deferred.

**Comment by MLGTASTICa** [comment]
> > Most marine bio armour sits at 35%-50% , administering 3.5U neurotox or 4.5U neurotox per spit seems way too much and might make sentinels/spitter/praetorians too suppresive for anyone that might try to push,might be a good idea to tone it down to 1U per spit at weakest and 3U at strongest.
> > Or you could make it more obvious that people are under the effect of neurotox and let them remove it by simple means like drinking water, making dylovene 10 times more effective against neurotox or making it metabolize away faster if marines are resting.
> 
> Given the hallucinogenic overlay, it's pretty obvious when you're affected by neurotox. I'm considering revising this so the druggy overlay doesn't happen until the neurotox has graduated to stage 2, using eyeblur accumulation to make it apparent at stage 1 instead.
> 
> At 35% bio armor, only the Prae is going to come close to 5U+ neuro, but yeah, I'm open to tweaking the values downward. Keep in mind that neuro metas fast, 0.4 / sec, and its effects are very slow acting and deferred, and there are no tackles to force stamina disable

Those are good points , however keep in mind that reaching the overdose threshold is far easier with spits since they have a low delay. My only worry is that marines will have too much neurotox in themselves to do anything after some light combatting and that will encourage hiding in FOB more..

**Comment by Surrealaser** [comment]
@MLGTASTICa Going over the armors, 50% bio resist actually appears to be the standard/baseline, so I'll work on that basis

**Comment by Chaznoodles** [comment]
Fun fact, 10u Neurotox might put you down, and 15 will. When you're down, you _stay_ down for a long time. Hello, easier captures despite capture gameplay being removed. That means 3-4 prae spits and you're done, 4-6ish queen spits and you're done, and so on. Even if you don't down them which will inevitably result in a kill, they're still absolutely crippled and unable to do much until the effects wear off. Far easier than killing and turns spitcastes into even more of a powerhouse than they already are - and that's not even including the effects of the gas.

You've made the light neurogas cause coughing, resulting in people dropping their weapons.

Making full neurogas extinguish fire/burning mobs means flaming the cloud to catch unaware xenos/protect the FOB is now pointless. Acid sprays can already extinguish xenos, so this feels like powercreep to make defences less useful.

Undocumented change:
image
Upped the accurate range on neuro spit, meaning you can now cross-screen people with it rather than it being a close-range risky option.

**Comment by Surrealaser** [comment]
@Chaznoodles Given that tackles no longer exist, it takes a long time to get dropped by neuro (i.e. take on 150+ Stamina damage), even assuming you just keep getting hit and focus fired, and don't rotate, use dylovene, purgers, etc. Unlike the old Prae spit, which would in fact down you straight up in 3-4 hits straight up without deferral or delay, and you're immersed in a disabling dose, this will take upwards of a minute to gnaw away at your stamina to this point, without any intervention, in order to just barely disable a target: (150-(2*20 ticks )/6 )= 18.33 ticks at Stage 2 + 20 Stage 1 ticks  = 38.33 ticks = 76.66 seconds = 1 minute, 16.66 seconds. The 10 stamina damage you take on hit post-armor will push this down somewhat, but in general it's going to be a minute upwards. Also 15U is just barely enough to drop (0.4 meta rate = 2.5 ticks per U = 37.5 ticks
15U takes no less than 4 hits to achieve from an Ancient Praetorian assuming 50% reduction... but because an Ancient Praetorian can only fire once per 2 seconds, it's actually 5 hits that are required assuming you don't linger in the gas (and 10 seconds of continuous, sustained fire on the same target without any pauses or delays; good luck achieving that).

Also coughing doesn't force weapon drop; it is purely a cosmetic emote.

As to the neurogas extinguishing fire, the point was to make neurogas competitive with the boiler's acid gas, which is most certainly isn't at this point in time; boilers can either choose between scoring significant lingering AoE damage, or providing safer cover. I might end up creating a 'medium' category for the Defiler's neurogas which doesn't extinguish, but I'll see how things pan out in test merge.

The effective range change was actually a test value I forgot to normalize on upload.

And yes, it is more tactically interesting, because instead of just dropping a dude in 4 hits immediately, neurospit users now have area denial and flush out tools that are in turn subject to counters accessible to the marines in the form of gas masks and antitoxins, while Boilers actually have an interesting choice to make when selecting globs.

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -29,36 +29,37 @@
...
 		adjustToxLoss(0.25)
 		adjustStaminaLoss(3)
-		blur_eyes(2)	
+		blur_eyes(2)
 	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_XENO_ACID))
 		adjustOxyLoss(4 + S.strength * 2)
 	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_XENO_NEURO))
 		if(!is_blind(src) && has_eyes())
 			to_chat(src, "<span class='danger'>Your eyes sting. You can't see!</span>")
+		if(!istype(S, /obj/effect/particle_effect/smoke/xeno/neuro/light)) //Only full neurogas blinds
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/carbon_defense.dm:38]: Snowflake behavior

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -29,36 +29,37 @@
...
-		reagents.add_reagent(/datum/reagent/toxin/xeno_neurotoxin, 5 + S.strength * 2)
+		reagents.add_reagent(/datum/reagent/toxin/xeno_neurotoxin, GAS_INHALE_REAGENT_TRANSFER_AMOUNT * S.strength)
 		if(prob(10 * S.strength)) //Likely to momentarily freeze up/fall due to arms/hands seizing up
 			to_chat(src, "<span class='danger'>You feel your body going numb and lifeless!</span>")
 	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_XENO_HEMODILE))
-		reagents.add_reagent(/datum/reagent/toxin/xeno_hemodile, 5 + S.strength * 2)
+		reagents.add_reagent(/datum/reagent/toxin/xeno_hemodile, GAS_INHALE_REAGENT_TRANSFER_AMOUNT * S.strength)
 	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_XENO_TRANSVITOX))
-		reagents.add_reagent(/datum/reagent/toxin/xeno_transvitox, 5 + S.strength * 2)
+		reagents.add_reagent(/datum/reagent/toxin/xeno_transvitox, GAS_INHALE_REAGENT_TRANSFER_AMOUNT * S.strength)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/carbon_defense.dm:47]: Undocumented change

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 3]
```diff
@@ -105,7 +105,7 @@
 		smoke_range = 3
 	else if(X.selected_reagent == /datum/reagent/toxin/xeno_transvitox)
 		N.smoke_type = /obj/effect/particle_effect/smoke/xeno/transvitox
-		N.strength = -0.75
+		N.strength = 0.75
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:108]: undocumented change

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -1330,70 +1330,140 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	accuracy_var_low = 3
 	accuracy_var_high = 3
 	bullet_color = COLOR_LIME
+	var/datum/effect_system/smoke_spread/xeno/smoke_system
+	var/list/datum/reagent/spit_reagents = new/list()
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1334]: Dont do new list

File: code/modules/projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -1330,70 +1330,140 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	var/reagent_transfer_amount
+	var/stagger_stacks
+	var/slowdown_stacks
+	var/smoke_strength
+	var/smoke_range
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1339]: Undocumented vars

File: code/modules/projectiles/ammo_datums.dm [hunk 6]
```diff
@@ -1330,70 +1330,140 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
-	damage = 45
+	damage = 20
+	stagger_stacks = 1
+	slowdown_stacks = 1
+	smoke_strength = 0.5
+	smoke_range = 0
+	reagent_transfer_amount = 5
+
+/datum/ammo/xeno/toxin/proc/set_reagents()
+	spit_reagents = list(/datum/reagent/toxin/xeno_neurotoxin = reagent_transfer_amount)
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1359]: Why

File: code/modules/projectiles/ammo_datums.dm [hunk 7]
```diff
@@ -1330,70 +1330,140 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 	accuracy_var_high = 3
-	damage = 45
+	damage = 20
+	stagger_stacks = 1
+	slowdown_stacks = 1
+	smoke_strength = 0.5
+	smoke_range = 0
+	reagent_transfer_amount = 5
+
+/datum/ammo/xeno/toxin/proc/set_reagents()
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1358]: undocumented proc

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by TiviPlus** [comment]
This PR is unatomized and has undocumented changes, and is a balance change during the freeze
Additionally most people disagree with the central aspect of this PR so split up this PR and or reconsider your neuro balance

**Pull request closed without merging**